### DISCUSSION TITLE
file associations

### DISCUSSION THREAD

**Discussion by ennerf** [hydraulic-software/conveyor#20] [Q&A] [answered]
Is there a way to create file type associations? Something similar to jpackage's [`--file-associations` ](https://docs.oracle.com/en/java/javase/17/jpackage/support-application-features.html)

**Comment by mikehearn** [comment, marked as answer]
Unfortunately, de-pain-ifying file associations and URL handlers is a feature that got cut for the first MVP release. There is a way to do it though, because the config is sufficiently flexible to add the platform specific metadata needed. There might be an issue on Linux where you need to drop a .mime file into a directory that's outside your app directory: we need to test it.

The first step is to see how far we can get with pure config in current releases. The resulting config might not be very pretty but OK, whatever. Once that's working we can define a high level declarative config and implement support in the core product, which should also enable things like icons per-file type.

@ogpublic Internal bug link is [CO-21](https://youtrack.hq.hydraulic.software/issue/CO-21).
